Add tests tree view after sorting

LeafShi1 opened the pull request:
<!-- Please read CONTRIBUTING.md before submitting a pull request -->

Fixes #10397


## Proposed changes

- Add unit tests "Remove_MultipleLevelsNodes_AfterSort", "Clear_MultipleLevelsNodes_AfterSort" and "Verify_NodeValue_AfterSortAndRemove" to test TreeView after sorting

<!-- We are in TELL-MODE the following section must be completed -->

## Risk

- Minimal

<!-- end TELL-MODE -->


## Test methodology <!-- How did you ensure quality? -->

- Unit tests


## Test environment(s) <!-- Remove any that don't apply -->

- .net 9.0.0-alpha.1.23578.3


<!-- Mention language, UI scaling, or anything else that might be relevant -->


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10401)

Tanya-Solyanik commented:
Do we have tests that add/remove tree nodes by `key` value - https://github.com/dotnet/winforms/blob/d906b5de936a8abb8670bd88a13aa9c046631d21/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs#L89

LeafShi1 commented:
> Do we have tests that add/remove tree nodes by `key` value -
> 
> https://github.com/dotnet/winforms/blob/d906b5de936a8abb8670bd88a13aa9c046631d21/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs#L89

Not yet. 

> Do we have tests that add/remove tree nodes by `key` value -
> 
> https://github.com/dotnet/winforms/blob/d906b5de936a8abb8670bd88a13aa9c046631d21/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs#L89

Not before, I updated the PR and added a test about Key in case "Verify_NodeValue_AfterSortAndRemove".
